feat: write types, add main layoutes

DaniilShuliyatev opened the pull request.

DaniilShuliyatev committed:
feat: write types, add main layoutes

Yauhenia-Zhdanovich commented on frontend/.gitignore:
```diff
@@ -0,0 +1,3 @@
+node_modules
+package-lock.json
```
не игнорим `package-lock.json`
https://blog.nashtechglobal.com/why-should-we-commit-package-lock-json-to-source-control/

Yauhenia-Zhdanovich commented on frontend/src/app/providers/ErrorBoundary/ui/ErrorBoundary.tsx:
```diff
@@ -0,0 +1,37 @@
+  children: ReactNode;
+}
+
+interface State {
```
везде используем типы(особенно в контексте приложения на реакте без классов, + типы более гибкие)

Yauhenia-Zhdanovich commented on frontend/src/app/providers/ErrorBoundary/ui/ErrorBoundary.tsx:
```diff
@@ -0,0 +1,37 @@
+import React, { type ReactNode, type ErrorInfo, Suspense } from "react";
+import { PageError } from "../../../../widgets/PageError";
```
в директории закинуть index.ts-ы, чтобы пути были красивые

`import { PageError } from "../../../../widgets";`

Yauhenia-Zhdanovich commented on frontend/src/app/providers/ErrorBoundary/ui/ErrorBoundary.tsx:
```diff
@@ -0,0 +1,37 @@
+  hasError: boolean;
+}
+
+export class ErrorBoundary extends React.Component<Props, State> {
```
`ErrorBoundary`
👍👍👍

Yauhenia-Zhdanovich commented on frontend/src/app/providers/router/ui/RequireAuth.tsx:
```diff
@@ -0,0 +1,19 @@
+import { Navigate, useLocation } from "react-router";
+import { RoutePaths } from "../../../../shared/config/routeConfig/routeConfig";
+
+export function RequireAuth({ children }: { children: JSX.Element }) {
```
```suggestion
export function RequireAuth({ children }: PropsWithChildren) {
```
определяемся, что будем использовать 

```suggestion
const RequireAuth: FC<PropsWithChildren> = ({ children } => {
```
и везде пользуем одинаково

Yauhenia-Zhdanovich commented on frontend/src/app/providers/router/ui/RequireAuth.tsx:
```diff
@@ -0,0 +1,19 @@
+  if (!auth) {
+    return (
+      <Navigate
+        to={RoutePaths["statistics"]}
```
```suggestion
        to={RoutePaths[AppRoutes.STATISTICS]}
```

Yauhenia-Zhdanovich commented on frontend/src/entity/PlayerCard/ui/PlayerCard.tsx:
```diff
@@ -0,0 +1,96 @@
+import { PlayerGamesTabel } from "./PlayerTabels/PlayerGamesTabel";
+import { PlayerGoalsTabel } from "./PlayerTabels/PlayerGoalsTabel";
+import { PlayerPassesTabel } from "./PlayerTabels/PlayerPassesTabel";
+import { PlayerPenaltyTabel } from "./PlayerTabels/PlayerPenaltyTabel";
```
`PlayerGamesTabel`
эти все компоненты надо переименовать, опечатка
`PlayerGamesTable`
если пользуешься VSCode, можно поставить [вот такой](https://marketplace.visualstudio.com/items/?itemName=streetsidesoftware.code-spell-checker) экстеншн, чекает спеллинг

Yauhenia-Zhdanovich commented on frontend/src/entity/PlayerCard/ui/PlayerCard.tsx:
```diff
@@ -0,0 +1,96 @@
+            </LeagueBlock>
+            <LeagueBlock sx={{ marginTop: "8px" }}>
+              <Avatar
+                sx={{ width: "18px", height: "18px", marginRight: "5px" }}
```
`sx={{ width: "18px", height: "18px", marginRight: "5px" }}`
давай стили все-же складывать к стилям, в 
`PlayerCard.style.tsx`
так попроще читать, когда они все вместе лежат

Yauhenia-Zhdanovich commented on frontend/src/entity/PlayerCard/model/types/player.ts:
```diff
@@ -0,0 +1,80 @@
+import { z } from "zod";
+
+const PlayerSchema = z.object({
```
`backend/src/schemas/playerSchemas.ts`
тут на беке лежит схема, можно посмотреть, какие данные будут приходить

Yauhenia-Zhdanovich commented on frontend/src/entity/PlayerCard/ui/PlayerCard.tsx:
```diff
@@ -0,0 +1,96 @@
+import { PlayerPenaltyTabel } from "./PlayerTabels/PlayerPenaltyTabel";
+import type { FC } from "react";
+
+export const PlayerCard: FC = () => {
```
в скоупе этого пр-а было бы хорошо не использовать мок непосредственно в компоненте, а передавать данные через пропсы и сразу использовать тип `export type Player = z.infer<typeof PlayerDataSchema> 
`

Yauhenia-Zhdanovich commented on frontend/src/entity/TeamCard/ui/TeamCardStatistics/TeamCardStatistics.style.tsx:
```diff
@@ -0,0 +1,71 @@
+
+export const CardWrapper = styled(Box)(() => ({
+  padding: "16px",
+  backgroundColor: "#FAFAFA",
```
это давай забирать из [материала](https://mui.com/material-ui/customization/color/#picking-colors) цвета

Yauhenia-Zhdanovich commented on frontend/src/entity/TeamCard/ui/TeamCard/TeamCard.tsx:
```diff
@@ -0,0 +1,60 @@
+  number: number;
+}
+
+export const TeamCard: FC<TeamCardProps> = (props) => {
```
```suggestion
export const TeamCard: FC<TeamCardProps> = ({number}) => {
```

Yauhenia-Zhdanovich commented on frontend/src/entity/TeamCard/ui/TeamCardsList/TeamCardsList.tsx:
```diff
@@ -0,0 +1,14 @@
+  return (
+    <>
+      {resp.map((item, i) => (
+        <TeamCard number={i + 1} key={item} />
```
это можно не выносить в отдельный компонент + если это список, логично сделать его через `<ul></ul>`

Yauhenia-Zhdanovich commented on frontend/src/widgets/Navbar/ui/Navbar.tsx:
```diff
@@ -0,0 +1,67 @@
+import ExitToAppOutlinedIcon from "@mui/icons-material/ExitToAppOutlined";
```
```
Navbar/
     index.tsx
     ui/
          Navbar.ts
          index.ts
```

```

Navbar/
     Navbar.ts
     index.ts
     /some folders if needed ??

```
давайте лучше так, мне удобнее смотреть👀

DaniilShuliyatev replied:
А нужен ли тогда public API (index.ts) файл еще и внутри Navbar folder если инкапсуляция импортов по сути будет происходить на уровне обращения к сущности widget, entity и т.д. ?

Yauhenia-Zhdanovich commented on frontend/src/pages/statisticsPage/StatisticsPage.tsx:
```diff
@@ -0,0 +1,68 @@
+
+const StatisticsPage: FC = () => {
+  const [teamsActive, setTeamsActive] = useState(true);
+  const [playersActive, setPlayersActive] = useState(false);
```
```suggestion
 enum StatisticsPageEnum {
  TEAMS = "TEAMS",
  PLAYERS = "PLAYERS",
}

const StatisticsPage: FC = () => {
  const [activeTab, setActiveTab] = useState(StatisticsPageEnum.TEAMS);

  const onTabChange = (tab: StatisticsPageEnum) => setActiveTab(tab);

  return (
    <>
      <ButtonGroup sx={{ marginTop: "31px" }}>
        <SwitchButton
          isActive={activeTab === StatisticsPageEnum.TEAMS}
          onClick={() => onTabChange(StatisticsPageEnum.TEAMS)}
          variant="outlined"
        >
          TEAMS
        </SwitchButton>
```
положить значения в enum, оставить только один колбек(сложная логика получается с двумя стейтами сразу)

Yauhenia-Zhdanovich commented on frontend/src/pages/statisticsPage/ui/StatisticsPage.tsx:
```diff
@@ -0,0 +1,57 @@
+        <SwitchButton
+          onClick={onPlayers}
+          variant="outlined"
+          isActive={playersActive}
```
https://stackoverflow.com/questions/76935768/react-does-not-recognize-the-isactive-prop-on-a-dom-element

Yauhenia-Zhdanovich commented on frontend/src/shared/consts/filteringYears.ts:
```diff
@@ -0,0 +1,12 @@
+export function filteringYears() {
```
`backend/src/routes/seasonRoutes.ts`
на сезоны будет endpoint, можно просто захардкодать пока массив

Yauhenia-Zhdanovich commented on frontend/src/shared/index.ts:
```diff
@@ -0,0 +1,5 @@
+
+export { CustomDropdownMenu } from './ui/CustomDropdownMenu/CustomDropdownMenu';
+export { RoutePaths, type AppRoutesProps, routeConfig, AppRoutes } from './config/routeConfig/routeConfig';
```
давай ВЕЗДЕ поправим импорты:)

Yauhenia-Zhdanovich commented on frontend/src/entity/TeamCard/ui/TeamCardStatistics/TeamStatistics.tsx:
```diff
@@ -0,0 +1,54 @@
+              maxHeight: "36px",
+            }}
+          >
+            <CustomTableCell>Played games</CustomTableCell>
```
давай тут map-ом сделаем тут

Yauhenia-Zhdanovich commented on frontend/src/entity/TeamCard/ui/TeamCardStatistics/TeamStatistics.tsx:
```diff
@@ -0,0 +1,54 @@
+      <Table aria-label="simple table">
+        <CustomTableHead>
+          <TableRow
+            sx={{
```
стили в styles ВЕЗДЕ

Yauhenia-Zhdanovich commented on frontend/src/entity/TeamCard/ui/TeamCardStatistics/TeamCardStatistics.style.tsx:
```diff
@@ -0,0 +1,97 @@
+  marginTop: "16px",
+  display: "flex",
+  flexDirection: "column",
+  border: "1px solid #E0E0E0",
```
import { grey } from '@mui/material/colors'
.... 
```suggestion
  border: `1px solid ${grey[300]}`,
```

Yauhenia-Zhdanovich commented on frontend/src/pages/statisticsPage/StatisticsPage.tsx:
```diff
@@ -0,0 +1,68 @@
+  const [teamsActive, setTeamsActive] = useState(true);
+  const [playersActive, setPlayersActive] = useState(false);
+
+  const onTeams = () => {
```
```suggestion
  const onTeamsClick = () => {
```
имя функции должно быть с глаголом

DaniilShuliyatev committed:
refactor: corrections based on PR review comments
fix: fix .gitignore file
refactor: add public api files, remove ui paths in wigets
refactor: import typeSchemas from backend
fix: fix PlayerTables files spelling
types: replaced interfaces with types
refactor: declaration components in single codestyle
refactor: remove TeamCardsList component, PlayerCard get data from props
style: add customColors endpoint in theme, add styles in PlayerCard, TeamCard, TeamCardStatistics, PageError, Navbar
refactor: update import urls

DaniilShuliyatev committed:
refactor: corrections based on second PR review comments
refactor StatisticsPage active tab logic
fix broser error on isActive external attribute
replace filteringYears function with filteredYears constant
move table headers cells to the constant level, these cells are now drowing using a map in the tables
ALL styles are now in .styles files
replace hex grey with MUI color
add nounused imports rull in biom config
refactor event callback names in CustomDropdownMenu component

Yauhenia-Zhdanovich commented on frontend/src/app/App.styles.ts:
```diff
@@ -0,0 +1,9 @@
+import { Box } from "@mui/material";
+import styled from "styled-components";
+
+
```
пустая строка

Yauhenia-Zhdanovich commented on frontend/src/app/App.styles.ts:
```diff
@@ -0,0 +1,9 @@
+    '&&': {
+        padding: "0 180px"
+    }
+}));
```
trailing newline eof

Yauhenia-Zhdanovich commented on frontend/src/app/App.tsx:
```diff
@@ -0,0 +1,16 @@
+import { Navbar } from "../widgets";
+import { AppRouter } from "./providers/router";
```
```suggestion
import { AppRouter } from "./providers";
```

Yauhenia-Zhdanovich commented on frontend/src/entity/PlayerCard/model/types/player.ts:
```diff
@@ -0,0 +1,4 @@
+import type { z } from "zod";
+import type { PlayerWithStatsSchema } from "../../../../../../backend/src/schemas/playerSchemas";
```
> backend/src/schemas/playerSchemas.ts
> тут на беке лежит схема, можно посмотреть, какие данные будут приходить

посмотреть, но оставить свою схему

Yauhenia-Zhdanovich commented on frontend/src/entity/PlayerCard/ui/PlayerCard.tsx:
```diff
@@ -0,0 +1,86 @@
+  CustomPlayerAvatar,
+  CustomCardContent,
+} from "./PlayerCard.style";
+import { PlayerCardsTable } from "./PlayerTables/PlayerCardsTable";
```
```suggestion
import { PlayerCardsTable, PlayerDuelsTable,  PlayerGamesTable ...} from "./PlayerTables";
```

Yauhenia-Zhdanovich commented on frontend/src/entity/PlayerCard/ui/PlayerTables/PlayerCardsTable.tsx:
```diff
@@ -0,0 +1,34 @@
+import type { FC } from "react";
+import { playerCardsHeaderCells } from "../../../../shared";
+
+export const PlayerCardsTable: FC = () => {
```
delete return

Yauhenia-Zhdanovich commented on frontend/src/entity/TeamCard/ui/TeamCardStatistics/TeamStatistics.tsx:
```diff
@@ -0,0 +1,35 @@
+          </CustomTableRow>
+        </CustomTableHead>
+        <CustomTableBody>
+          <CustomTableRow>
```
очень похожие таблицы, может их можно как-то объединить

Yauhenia-Zhdanovich commented on frontend/src/pages/statisticsPage/StatisticsPage.tsx:
```diff
@@ -0,0 +1,69 @@
+  return (
+    <>
+      <CustomButtonGroup>
+        <SwitchButton
```
https://mui.com/material-ui/react-tabs/#full-width

Yauhenia-Zhdanovich commented on frontend/src/shared/ui/CustomDropdownMenu/CustomDropdownMenu.tsx:
```diff
@@ -0,0 +1,72 @@
+  filteringValues: number[];
+};
+
+export const CustomDropdownMenu: FC<DropdownMenuProps> = (props) => {
```
```suggestion
export const CustomDropdownMenu: FC<DropdownMenuProps> = ({
  width, labels, filteringValues
}) => {
```

Yauhenia-Zhdanovich replied:
все еще актуально 
вариант 
```suggestion
export const CustomDropdownMenu: FC<DropdownMenuProps> = ({ width, labels, filteringValues }) => {
```
читается лучше, чем 
`  const { width, Labels, filterValues } = props;`

Yauhenia-Zhdanovich commented on frontend/src/shared/ui/CustomDropdownMenu/CustomDropdownMenu.tsx:
```diff
@@ -0,0 +1,72 @@
+import { useState, type FC } from "react";
+import { CustomFormControl } from "./CustomDropdownMenu.styles";
+
+interface Label {
```
interface + "неговорящее" имя интерфейса, особенно если учесть, что он довольно специфичен
https://www.freecodecamp.org/news/how-to-write-better-variable-names/

Yauhenia-Zhdanovich commented on frontend/src/shared/ui/CustomDropdownMenu/CustomDropdownMenu.tsx:
```diff
@@ -0,0 +1,72 @@
+  return (
+    <CustomFormControl width={width}>
+      <InputLabel id="demo-simple-select-label">{label}</InputLabel>
+      <Select
```
https://mui.com/material-ui/react-autocomplete/

Yauhenia-Zhdanovich commented on frontend/src/shared/ui/CustomDropdownMenu/CustomDropdownMenu.tsx:
```diff
@@ -0,0 +1,72 @@
+    <CustomFormControl width={width}>
+      <InputLabel id="demo-simple-select-label">{label}</InputLabel>
+      <Select
+        labelId="demo-simple-select-label"
```
в константу

Yauhenia-Zhdanovich commented on frontend/src/widgets/Navbar/Navbar.tsx:
```diff
@@ -0,0 +1,60 @@
+          <CustomIcon />
+        </LogoutButton>
+      </Header>
+      <Navigation>
```
https://mui.com/material-ui/react-tabs/#introduction

Yauhenia-Zhdanovich commented on frontend/src/app/providers/router/RequireAuth.tsx:
```diff
@@ -0,0 +1,23 @@
+import type { FC } from "react";
+
+type RequireAuthProps = {
+  children: React.ReactNode;
```
`PropsWithChildren`

Yauhenia-Zhdanovich replied:
```
import type { PropsWithChildren } from 'react'

export const RequireAuth: FC<PropsWithChildren> ...
```

Yauhenia-Zhdanovich commented on frontend/src/widgets/PageError/PageError.tsx:
```diff
@@ -0,0 +1,18 @@
+  const reloadPage = () => {
+    location.reload();
+  };
+  return (
```
trailing newline before return

Yauhenia-Zhdanovich commented on frontend/src/widgets/Navbar/Navbar.style.ts:
```diff
@@ -0,0 +1,60 @@
+  cursor: "default",
+}));
+
+// Стили для Navigation
```
?? удаляем бесполезные комменты

Yauhenia-Zhdanovich commented on frontend/src/app/providers/theme/theme.ts:
```diff
@@ -0,0 +1,26 @@
+      contrastText: "#0B79D0"
+    }
+  },
+});
```
```
type CustomTheme = typeof theme;

declare module 'styled-components' {
  export interface DefaultTheme extends CustomTheme { }
}

```

DaniilShuliyatev committed:
refactor: corrections based on third PR review comments
add missing trailing newlines in each file
replaced the statistics page RoutePath from / to /statistics
set the open vite server property to /statistics
refactor imports: add additional index.ts files + rewrote the imports of the "./data/data" format to "./data"
remove extra empty lines between imports and component
replace extra returns with parentheses
refactor types naming in ErrorBoundary, RequireAuth and in CustomDropdownMenu, props naming in TeamCard and PlayerCard components
the TeamSchema and PlayerSchema types are written and correspond to the types on the backend
the theme variable is passed as props inside styled component instead of being imported into the .styles file
add new CustomTable component that replaced all files with table components
replace custom navigation buttons in StstisticsPage with Tabs from MUI
add new TabelPanel UI component for temporary visualization of the Tabs operation display
refactor Select component in CustomDropdownMenu with Autocomplete component from MUI
remove all useless comments
refactor the logic of displaying the active tab and replace custom Navigation components with Tabs MUI components in the Navbar component
add missing empty line before returns

Yauhenia-Zhdanovich commented on frontend/src/entity/PlayerCard/ui/PlayerCard.tsx:
```diff
@@ -0,0 +1,153 @@
+    setCardsBodyCells([
+      playerData.cards.yellow || "No info",
+      playerData.cards.yellowred || "No info",
+      playerData.cards.red || "No info",
```
```suggestion

const getDisplayString = (value: string) => value ?? 'No info'
    .....
    setCardsBodyCells([
      getDisplayString(playerData.cards.yellow),
      getDisplayString(playerData.cards.yellowred),
      getDisplayString(playerData.cards.red),
```
или

```suggestion

    Object.values(playerData.cards).map(getDisplayString)
    setCardsBodyCells(Object.values(playerData.cards).map(getDisplayString));
```

Yauhenia-Zhdanovich commented on frontend/src/entity/PlayerCard/ui/PlayerCard.tsx:
```diff
@@ -0,0 +1,153 @@
+type TableCells = (string | number)[];
+
+export const PlayerCard: FC<PlayerCardProps> = ({ playerData }) => {
+  const [cardsBodyCells, setCardsBodyCells] = useState<TableCells>([]);
```
нужен ли тут `useState`? значения в этом компоненте не изменяются + родительский компонент, скорее всего, не будет мутировать объекты player

Yauhenia-Zhdanovich commented on frontend/src/app/index.ts:
```diff
@@ -0,0 +1,4 @@
+export { ErrorBoundary } from "./providers/ErrorBoundary";
+export { AppRouter } from "./providers/router";
+export { theme } from "./providers/theme/theme";
```
☹️

Yauhenia-Zhdanovich commented on frontend/src/pages/statisticsPage/StatisticsPage.style.ts:
```diff
@@ -0,0 +1,44 @@
+  leftBorder?: boolean;
+};
+
+export const CustomTab = styled(Tab, {
+  shouldForwardProp: (prop) => prop !== "leftBorder",
+})<CustomTabProps>(({ leftBorder }) => ({
+  "&&": {
+    opacity: 1,
+    fontSize: "13px",
+    lineHeight: "22px",
+    paddingBottom: "4px",
+    paddingTop: "4px",
+    minHeight: "30px",
+    borderLeft: leftBorder ? "1px solid #2196F3" : null,
+  },
+}));
```
```suggestion
export const CustomTab = styled(Tab
)(() => ({
  "&&": {
    opacity: 1,
    fontSize: "13px",
    lineHeight: "22px",
    paddingBottom: "4px",
    paddingTop: "4px",
    minHeight: "30px",
  },
  "&:first-of-type": {
    borderRight: "1px solid #2196F3",
  },
}));```

Yauhenia-Zhdanovich commented on frontend/src/pages/statisticsPage/StatisticsPage.style.ts:
```diff
@@ -0,0 +1,44 @@
+export const CustomTabs = styled(Tabs)(() => ({
+  maxWidth: "220px",
+  minHeight: "30px",
+  color: "#2196F3",
```
```suggestion
import { colors } from '@mui/material'

  color: colors.blue[500],
```

Yauhenia-Zhdanovich commented on frontend/src/pages/statisticsPage/StatisticsPage.tsx:
```diff
@@ -0,0 +1,75 @@
+
+const resp = ["data1", "data2", "data3", "data4", "data5"];
+
+enum StatisticsPageEnum {
+  TEAMS = 0,
+  PLAYERS = 1,
+}
```
```suggestion
enum StatisticsPageEnum {
  TEAMS,
  PLAYERS,
}
```

Yauhenia-Zhdanovich commented on frontend/src/pages/statisticsPage/StatisticsPage.tsx:
```diff
@@ -0,0 +1,75 @@
+    newValue: StatisticsPageEnum,
+  ) => {
+    setActiveTab(newValue);
+  };
```
```suggestion
 const handleTabChange = (
    _: unknown,
    newValue: StatisticsPageEnum,
  ) => {
    setActiveTab(newValue);
  };
```

Yauhenia-Zhdanovich commented on frontend/src/shared/ui/CustomDropdownMenu/CustomDropdownMenu.tsx:
```diff
@@ -0,0 +1,59 @@
+type DropdownMenuLabel = {
+  first: string;
+  second: string;
+};
```
зачем тут 2 лейбла?

DaniilShuliyatev replied:
По макету, в пустом dropdownMenu один label, а при наличии выбранного значения - label уже другой

Yauhenia-Zhdanovich commented on frontend/src/widgets/Navbar/Navbar.style.ts:
```diff
@@ -0,0 +1,49 @@
+
+export const CustomTabs = styled(Tabs)(() => ({
+  minHeight: "48px",
+  color: "#2196F3",
```
по всему css коду использовать [переменные](https://mui.com/material-ui/customization/color/) из MaterialUI вместо строк

Yauhenia-Zhdanovich commented on frontend/src/widgets/Navbar/Navbar.tsx:
```diff
@@ -0,0 +1,57 @@
+enum NavbarTabsEnum {
+  STATISTICS_PAGE = 0,
+  TEAM_PAGE = 1,
+}
```
```suggestion
enum NavbarTabsEnum {
  STATISTICS_PAGE,
  TEAM_PAGE,
}
```

Yauhenia-Zhdanovich commented on frontend/src/shared/ui/CustomTable/CustomTable.tsx:
```diff
@@ -0,0 +1,38 @@
+      <CustomTableHead>
+        <CustomTableRow>
+          {headerCells.map((item) => (
+            <CustomTableCell key={nanoid()}>{item}</CustomTableCell>
```
использование `nanoid()` для ключа 
https://dev.to/bytebodger/comment/1df8g

Yauhenia-Zhdanovich commented on frontend/src/shared/ui/TabPanel/TabPanel.tsx:
```diff
@@ -0,0 +1,26 @@
+  dir?: string;
+  index: number;
+  value: number;
+};
```
```suggestion
type TabPanelProps = {
  dir?: string;
  index: number;
  value: number;
} & ChilderWithProps
```

Yauhenia-Zhdanovich commented on frontend/src/shared/config/routeConfig.tsx:
```diff
@@ -0,0 +1,33 @@
+  authOnly?: boolean;
+};
+
+export enum AppRoutes {
+  STATISTICS = "statistics",
+  FANTASY_TEAMS = "fantasy_teams",
+
+  NOT_FOUND = "not_found",
+}
```
```suggestion
export enum AppRoutes {
  STATISTICS = "statistics",
  FANTASY_TEAMS = "fantasy_teams",
  NOT_FOUND = "not_found",
}
```

DaniilShuliyatev replied:
Это, если что, не опечатка, я специально оставлял тут пустую строку, что бы явно отделить NOT_FOUND  роут от остальных, но если по стайл гайду так - не красиво, то исправлю

Yauhenia-Zhdanovich replied:
> я специально оставлял тут пустую строку
 
если очень хочется разделить, то уже лучше комментарий написать,  не очень понятно было, зачем пустая строка в `enum`

Yauhenia-Zhdanovich commented on frontend/src/entity/TeamCard/ui/TeamCard/TeamCard.tsx:
```diff
@@ -0,0 +1,48 @@
+          </MainText>
+          <CountryBlock>
+            <CustomCountryAvatar src={teamMockedData.league.country.flag} />
+            {teamMockedData.league.country.name}
```
передать `team` как props

Yauhenia-Zhdanovich commented on frontend/src/app/providers/theme/theme.ts:
```diff
@@ -0,0 +1,26 @@
+      main: "#2196F3",
+    },
+    error: {
+      main: "#F44336",
```
если это стандартный цвет из https://mui.com/material-ui/customization/color/, не имеет смысла хранить его в теме
![image](https://github.com/user-attachments/assets/7e1fe75e-c165-4742-afe3-13ace8900630)

Yauhenia-Zhdanovich commented on frontend/src/app/providers/router/AppRouter.tsx:
```diff
@@ -0,0 +1,26 @@
+      <Route
+        key={route.path}
+        path={route.path}
+        element={
```
```suggestion
  const renderWithWrapper = useCallback((route: AppRoutesProps) => {
    const element = (
      <Suspense fallback={<Skeleton>{route.element}</Skeleton>}>
        {route.element}
      </Suspense>
    );
   
    return (
      <Route
        key={route.path}
        path={route.path}
        element={
```

Yauhenia-Zhdanovich commented on frontend/src/entity/PlayerCard/index.ts:
```diff
@@ -0,0 +1,2 @@
+export { PlayerCard } from "./ui/PlayerCard";
```
```suggestion
export { PlayerCard } from "./ui";
```

DaniilShuliyatev committed:
replace nanoid library with uuid library for generating components unique keys
add missing index.ts files and update imports
replace children prop in types with built-in PropsWithChildren type in react
add missing empty line before return in AppRouter
remove MUI built-in colors from theme
add the DefaultTheme type with the CustomTheme type
replace colors in .style files with MUI color variables
replace useState with static variables and add a function to check the value for null in PlayerCard file
TeamCard component does not receive data directly, but through props
remove numeric values as value from StatisticsPage and Navbar enums
remove extra empty line from routeConfig
change DropdownLabels type keys naming and add props spreading in CustomDropdownMenu component
replace unused first argument variables in handle functions with _ unknown type

Yauhenia-Zhdanovich commented on frontend/src/shared/ui/CustomTable/CustomTable.tsx:
```diff
@@ -0,0 +1,38 @@
+      <CustomTableBody>
+        <CustomTableRow>
+          {bodyCells?.map((item) => (
+            <CustomTableCell key={uuidV4()}>{item}</CustomTableCell>
```
![image](https://github.com/user-attachments/assets/167a631e-e2f5-4122-8e0e-c2d672947bd4)
https://dev.to/bytebodger/comment/1df8g

DaniilShuliyatev committed:
refactor: PR review changes
replace two dynamic labels with one in CustomDropdownMenu component, add placeholder
remove dropdownLabels file from constants
return error main color to theme
replace uuid key generator in CustomTable with iteration elements value
